api-catalog-ui/frontend/cypress/e2e/graphql/graphql-apiml.cy.js

```diff
@@ -0,0 +1,258 @@
+
+        cy.get(textareaSelector).should('be.visible');
+
+        cy.get('#graphiql-session > div:nth-child(1) > div > div:nth-child(1) > section > div.graphiql-editor > div > div.CodeMirror-scroll > div.CodeMirror-sizer > div > div > div > div.CodeMirror-code > div > pre > span > span')
+            .should('exist')
+            .should('be.visible')
+            .and('have.text', '# Write your query here!');
+
+        const query = '{ "query": "{ hello }" }';
+        cy.get(textareaSelector)
+            .first()
+            .focus()
+            .type('{ctrl}a')
+            .type('{del}')
+            .type(query, { parseSpecialCharSequences: false });
+
+        cy.get('#graphiql-session > div:nth-child(1) > div > div:nth-child(1) > section > div.graphiql-editor > div > div.CodeMirror-scroll > div.CodeMirror-sizer > div > div > div > div.CodeMirror-code')
+            .then($container => {
+                const text = $container.text().trim();
+                expect(text).to.include('{ "query": "{ hello }" }');
+            })
+    })
+
+    it('Should run query', () => {
+        login()
+
+        cy.contains('Discoverable client with GraphQL').click();
+
+        const query = 'query {' +
+            'getAllBooks{' +
+            'name' +
+            ' }'
+            ;
+        const textareaSelector = '#graphiql-session > div:nth-child(1) > div > div:nth-child(1) > section > div.graphiql-editor > div > div:nth-child(1) > textarea';
+
+        cy.get(textareaSelector)
+            .first()
+            .focus()
+            .type('{ctrl}a')
+            .type('{del}')
+            .type(query, { parseSpecialCharSequences: false });
+
+        cy.get('#graphiql-session > div:nth-child(1) > div > div:nth-child(1) > section > div.graphiql-toolbar > button').click();
+
+        cy.get('span.cm-def').should('contain.text', 'data');
+
+        cy.get('#graphiql-session > div:nth-child(3) > div > section > div > div.CodeMirror-scroll > div.CodeMirror-sizer > div > div > div > div.CodeMirror-code')
+            .then($container => {
+                const text = $container.text().trim();
+                expectedKeyWords.forEach(word => {
+                    expect(text).to.include(word);
+                });
+            })
+    })
+
+    it('Should add and remove a tab in the playground', () => {
+        login()
+
+        cy.contains('Discoverable client with GraphQL').click();
+
+        cy.get('#graphiql-container > div > div.graphiql-main > div.graphiql-sessions > div.graphiql-session-header > div > button')
+            .click()
+
+        cy.get('#graphiql-session-tab-1')
+            .should('exist')
+            .should('contain.text', 'My Query 2');
+
+        cy.get('#graphiql-container > div > div.graphiql-main > div.graphiql-sessions > div.graphiql-session-header > ul > li.graphiql-tab.graphiql-tab-active > button.graphiql-un-styled.graphiql-tab-close')
```

Comment:
consider also using data-test label (you can find other examples in the existing code), to directly query the element avoiding long selectors.